lib/rules/object.js

```diff
@@ -93,6 +93,54 @@ class Keys extends Rule {
     }
 }
 
+class Pattern extends Rule {
+    compile(params) {
+        this._keyRegex = new RegExp(params.args[0]);
+        this._valueSchema = params.args[1];
+    }
+
+    validate(params, callback) {
+        const value = params.value;
+        if (value === undefined || value === null) {
+            return callback(this.error(params));
+        }
```

Comment:
I don't think we should bail here, the consumer might allow null or undefined values to be set as the property.

```
.pattern(/./, Jo.number().allow(null))
```

They can add `.required()` if they don't want to allow the values to be undefined